dowhy/causal_identifier.py

```diff
@@ -39,12 +45,19 @@ def __init__(self, graph, estimand_type,
         self._proceed_when_unidentifiable = proceed_when_unidentifiable
         self.logger = logging.getLogger(__name__)
 
-    def identify_effect(self, optimize_backdoor=False):
+    def identify_effect(self, optimize_backdoor=False, costs=None, conditional_node_names=None):
         """Main method that returns an identified estimand (if one exists).
 
         If estimand_type is non-parametric ATE, then  uses backdoor, instrumental variable and frontdoor identification methods,  to check if an identified estimand exists, based on the causal graph.
 
         :param self: instance of the CausalIdentifier class (or its subclass)
+        :param optimize_backdoor: if True, uses an optimised algorithm to compute the backdoor sets
+        :param costs: non-negative costs associated with variables in the graph. Only used
+        for estimand_type='non-parametric-ate' and method_name='efficient-mincost-adjustment'. If
+        no costs are provided by the user, and method_name='efficient-mincost-adjustment', costs
+        are assumed to be equal to one for all variables in the graph.
+        :param conditional_node_names: variables that are used to determine treatment. If none are
```

Comment:
I'm curious---what are these conditional_node_names graphically? are they parents of the treatment only?

Reply:
See my answer in the [comment](https://github.com/py-why/dowhy/pull/549#issuecomment-1188730376) below